src/site/controllers/WebController.php

```diff
@@ -89,15 +84,13 @@ protected static function getExtraParametersSpec(
     return shape('required' => ImmVector {}, 'optional' => ImmVector {});
   }
 
-  abstract public function getResponseAsync(
-    ResponseInterface $response,
-  ): Awaitable<ResponseInterface>;
+  abstract public function getResponseAsync(): Awaitable<Message\IResponse>;
 
   final protected function getRequestedPath(): string {
     return $this->request->getUri()->getPath();
   }
 
-  final protected function getRequestedHost(): string {
+  final protected function getRequestedHost(): ?string {
```

Comment:
Nuxed's `Uri::getHost()` returns `?string` as the host can be missing sometimes ( when its only a path )